projects/core/components/data-list/data-list.style.less

```diff
@@ -22,15 +23,29 @@ tui-data-list {
     &[data-list-size='s'] {
         /* stylelint-disable-next-line */
         --tui-data-list-margin: 0rem;
+
+        & > .t-empty {
+            .data-list-item('s');
+        }
+    }
+
+    &[data-list-size='m'] > .t-empty {
+        .data-list-item('m');
     }
 
     &[data-list-size='l'] {
         --tui-data-list-padding: 0.375rem;
         --tui-data-list-margin: 0.125rem;
+
+        & > .t-empty {
+            .data-list-item('l');
+        }
     }
 
     & > .t-empty {
-        margin: 0.75rem 1rem;
+        display: flex;
```

Comment:
I think we can move these 3 rules under the mixin as well, since they are also used in `tuiOption`.